Essentials/src/com/earth2me/essentials/signs/SignBuy.java

```diff
@@ -18,12 +17,23 @@ public SignBuy() {
     @Override
     protected boolean onSignCreate(final ISign sign, final User player, final String username, final IEssentials ess) throws SignException {
         validateTrade(sign, 1, 2, player, ess);
+        if(ess.getSettings().isBuySignUpdatedToWorth()) {
+            try {
+              updateFromWorth(sign, ess, ess.getSettings().getEcoBuyMultiplier());
+            }
+            catch (SignException e) {
+              // It's fine
+            }
+        }
```

Comment:
No, it's not fine?

Reply:
In this location this means that the price could not be found in worth.yml If we do nothing here the sign will work as it used to. Should I elaborate on that in the comment?

Reply:
There should be a check in the method to validate this, not a try catch here

Reply:
Could you help me out how I should do that? 
The solution in my head now is to pass a boolean whether or not an exception should be thrown. But I'm not fond if this idea. I'll have to know the difference between a 'create' update and a 'use' update if I want to be able to make the decision whether or not to throw the exception.

Reply:
In general, it's bad practice to throw exceptions during normal operation of a program. They're computationally expensive and make the code harder to understand when an exception is "supposed" to be thrown. You should return a value like a boolean that indicates whether the price changed, and then if the price _did_ change, cancel the buy/sell and let the player know about the price change so they can decide whether to complete their transaction at the new price.

Reply:
Fixed